File: quotes/src/main/java/se/callista/quotes/Quote.java
package se.callista.quotes;

import java.net.InetAddress;
import java.net.UnknownHostException;

public class Quote {

	private String ipAddress;

	private String quote;
	private String language;

	public Quote() {}

	public Quote(String quote, String language) {
		this.quote = quote;
		this.language = language;
		ipAddress = findMyIpAddress();
	}
	
	public String getQuote() {
		return quote;
	}
	public String getLanguage() {
		return language;
	}
	public String getIpAddress() {
		return ipAddress;
	}

	private String findMyIpAddress() {
		try {
			return InetAddress.getLocalHost().getHostAddress();
		} catch (UnknownHostException e) {
			throw new RuntimeException(e);
		}
	}
}

File: go-quotes/quote.go
package main

type Quote struct {
        HardwareArchitecture string `json:"hardwareArchitecture"`
        OperatingSystem string `json:"operatingSystem"`
        IpAddress string `json:"ipAddress"`
        Quote string `json:"quote"`
        Language string `json:"language"`
}

func (q *Quote) ToString() string {
        return q.HardwareArchitecture + " " + q.OperatingSystem + " " + q.IpAddress + " " + q.Language + " " + q.Quote
}